### ISSUE TITLE
[Common] tpl is extremely slow for large charts

### ISSUE THREAD

**Issue by ashleydavies** [bitnami/charts#17295]
### Name and Version

Common

### What is the problem this feature will solve?

The `tpl` function is extremely slow when linting/rendering/... large charts - particularly when deploying charts of charts used to aggregate many charts into one deployable monolith, since the way it works requires doing a full copy of the entire build context.

For instance, it takes several seconds to build each `bitnami` chart we use as part of a wider deployment of several third-party services.

### What is the feature you are proposing to solve the problem?

I've written a `fast_tpl` function internally which functions as a drop-in replacement for `tpl`, but avoids invoking `tpl` in the common case by inspecting the string passed, and skipping tpl if:

1. The string has no template visible within it
2. The string is a single template which can be trivially resolved through simple recursion, i.e. the string looks like `{{ .Values.a.b.c }}`

For us, this made our charts build, lint and render something on the order of 50-90% faster by avoiding the vast majority of `tpl` invocations.

I'd like to propose that, if I open-source this function, I integrate it into your common chart and hence significantly reduce the time to lint and build Bitnami charts using that.

To be clear - offering to do the work here, just looking for a sign of interest (i.e. would someone on your side be happy to review, help verify correctness, and ultimately land this change?) to avoid spending time opening a PR that isn't of interest if you're not keen on the idea.

**Comment by ashleydavies** [comment]
Actually, it looks like there's an open PR at https://github.com/helm/helm/pull/11351 which should solve `tpl` performance issues. I'll leave this open for now but I don't think this is high priority whilst that PR looks like it's close to landing.

**Comment by carrodher** [comment]
Thanks for your interest, I agree with your proposal but it's true that this topic seems something to be implemented in Helm itself... Apart from the existing PR, did you try to contribute there? Maybe your changes could complement that PR or even creating a new one

**Comment by ashleydavies** [comment]
Yeah, I've asked the author of that PR if they're done and hopefully if so re-prompt a review (I don't think anyone directly mentioned the maintainer to let them know that it's ready). Failing that I'll see if Helm is open to integrating my optimisation directly into itself, since it should be a drastic improvement in the common case.

I'll close for now, and re-open and get back to you if there's no luck pursuing that, if that sounds good?

Thanks for the reply!
